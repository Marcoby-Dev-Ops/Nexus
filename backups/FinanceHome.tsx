import React, { useMemo } from 'react';
import { DollarSign, CreditCard, Receipt, ArrowUpRight, FileText, Calculator } from 'lucide-react';
import { PageTemplates } from '@/components/patterns/PageTemplates';
import { ContentCard } from '@/components/patterns/ContentCard';
import { KpiCard } from '@/components/dashboard/KpiCard';
import { SimpleBarChart } from '@/components/dashboard/SimpleBarChart';
import { useDepartmentKPIs } from '@/lib/hooks/useDepartmentKPIs';
import { usePayPalTransactions } from '@/lib/hooks/usePayPalTransactions';
import type { PayPalTransaction } from '@/lib/hooks/usePayPalTransactions';

/**
 * @name FinanceHome
 * @description Finance department dashboard for managing invoices, expenses, and financial reports.
 * @returns {JSX.Element} The rendered FinanceHome component.
 */

// Placeholder expense data until wired to live source
const expenseData = [
  { name: 'Salaries', value: 45000 },
  { name: 'Marketing', value: 12000 },
  { name: 'Operations', value: 8500 },
  { name: 'Software', value: 5200 },
  { name: 'Office', value: 3200 },
];

const quickActions = [
  { label: 'New Invoice', icon: <Receipt className="w-5 h-5" />, onClick: () => console.log('New Invoice') },
  { label: 'Record Expense', icon: <CreditCard className="w-5 h-5" />, onClick: () => console.log('Record Expense') },
  { label: 'Generate Report', icon: <FileText className="w-5 h-5" />, onClick: () => console.log('Generate Report') },
  { label: 'Calculate Tax', icon: <Calculator className="w-5 h-5" />, onClick: () => console.log('Calculate Tax') },
];

// Helper for currency formatting
const currencyFmt = new Intl.NumberFormat('en-US', { style: 'currency', currency: 'USD' });

const FinanceHome: React.FC = () => {
  const { loading, metrics } = useDepartmentKPIs('finance');
  const { loading: txLoading, transactions } = usePayPalTransactions(10);

  type KpiItem = { title: string; value: string; delta?: string; trend?: string };

  const financeKpiData: KpiItem[] = useMemo(() => {
    const currencyFormatter = new Intl.NumberFormat('en-US', { style: 'currency', currency: 'USD' });

    const totalRevenueValue = metrics?.totalRevenue ?? 0;

    return [
      {
        title: 'Total Revenue',
        value: loading ? 'â€”' : currencyFormatter.format(totalRevenueValue),
        delta: undefined,
        trend: 'up',
      },
      { title: 'Outstanding Invoices', value: '$45,200', delta: '-8.3%', trend: 'down' }, // placeholder
      { title: 'Monthly Expenses', value: '$32,100', delta: '+5.2%', trend: 'up' }, // placeholder
      { title: 'Profit Margin', value: '32.8%', delta: '+2.1%', trend: 'up' }, // placeholder
    ];
  }, [loading, metrics]);

  const revenueData = metrics?.monthlyRevenue?.length ? metrics.monthlyRevenue : [
    { name: 'Jan', value: 0 },
  ];

  return (
    <PageTemplates.Department
      title="Finance Hub"
      subtitle="Manage invoices, expenses, and financial reports with automated reconciliation"
    >
      {/* Quick Actions */}
      <ContentCard title="Quick Actions" variant="elevated" className="mb-8">
        <div className="grid grid-cols-2 md:grid-cols-4 gap-4">
          {quickActions.map((action, index) => (
            <button
              key={index}
              onClick={action.onClick}
              className="flex flex-col items-center p-6 rounded-lg bg-background border border-border hover:border-primary hover:bg-primary/5 hover:shadow-md transition-all duration-200 group"
            >
              <div className="p-4 rounded-lg bg-primary/10 text-primary group-hover:bg-primary/20 transition-all duration-200 mb-3">
                {action.icon}
              </div>
              <span className="text-sm font-medium text-foreground group-hover:text-primary transition-colors duration-200">
                {action.label}
              </span>
            </button>
          ))}
        </div>
      </ContentCard>

      {/* KPI Cards */}
      <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-6 mb-8">
        {financeKpiData.map((kpi: KpiItem, index: number) => (
          <KpiCard key={index} {...kpi} />
        ))}
      </div>

      {/* Charts */}
      <div className="grid grid-cols-1 lg:grid-cols-2 gap-8 mb-8">
        <ContentCard 
          title="Revenue Trend" 
          variant="elevated"
          className="hover:shadow-xl transition-all duration-300"
          action={
            <button className="flex items-center space-x-2 px-4 py-4 bg-primary/5 text-primary hover:bg-primary/10 hover:text-primary/90 rounded-lg transition-all duration-200 font-medium border border-border">
              <span className="text-sm">View Details</span>
              <ArrowUpRight className="w-4 h-4" />
            </button>
          }
        >
          <div className="mb-6">
            <p className="text-sm text-muted-foreground">Monthly revenue performance</p>
          </div>
          <SimpleBarChart data={revenueData} />
        </ContentCard>

        <ContentCard 
          title="Expense Breakdown" 
          variant="elevated"
          className="hover:shadow-xl transition-all duration-300"
          action={
            <button className="flex items-center space-x-2 px-4 py-4 bg-primary/5 text-primary hover:bg-primary/10 hover:text-primary/90 rounded-lg transition-all duration-200 font-medium border border-border">
              <span className="text-sm">Manage Expenses</span>
              <ArrowUpRight className="w-4 h-4" />
            </button>
          }
        >
          <div className="mb-6">
            <p className="text-sm text-muted-foreground">Monthly expense categories</p>
          </div>
          <SimpleBarChart data={expenseData} />
        </ContentCard>
      </div>

      {/* Recent Transactions */}
      <ContentCard 
        title="Recent Transactions" 
        variant="elevated"
        action={
          <button className="px-4 py-4 bg-primary/5 text-primary hover:bg-primary/10 hover:text-primary/90 rounded-lg transition-all duration-200 text-sm font-medium border border-border">
            View All Transactions
          </button>
        }
      >
        <div className="space-y-4">
          {txLoading ? (
            <p className="text-sm text-muted-foreground">Loading...</p>
          ) : transactions.length === 0 ? (
            <p className="text-sm text-muted-foreground">No PayPal transactions found.</p>
          ) : (
            transactions.map((transaction: PayPalTransaction, index: number) => (
              <div key={index} className="flex items-center justify-between p-4 rounded-lg hover:bg-muted/50 transition-colors duration-200">
                <div className="flex items-center space-x-4">
                  <div className={`p-4 rounded-lg bg-success/10 text-success`}>
                    <DollarSign className="w-5 h-5" />
                  </div>
                  <div>
                    <h4 className="font-medium text-foreground">{transaction.txn_id}</h4>
                    <p className="text-sm text-muted-foreground">{new Date(transaction.captured_at).toLocaleDateString()}</p>
                  </div>
                </div>
                <div className={`font-semibold text-success`}>
                  {currencyFmt.format(transaction.amount)}
                </div>
              </div>
            ))
          )}
        </div>
      </ContentCard>
    </PageTemplates.Department>
  );
};

export default FinanceHome; 