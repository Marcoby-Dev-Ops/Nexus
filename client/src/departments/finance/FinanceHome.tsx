import { KpiCard } from '../../components/dashboard/KpiCard';
import { SimpleBarChart } from '../../components/dashboard/SimpleBarChart';
import React from 'react';

// Mock data
const financeSummary = {
  mtdRevenue: 12000,
  qtdRevenue: 34000,
  ytdRevenue: 120000,
  cashOnHand: 45000,
  expenses: 8000,
  netProfit: 4000,
};
const invoices = [
  { number: 'INV-001', client: 'Acme Corp', amount: 2500, status: 'Paid', due: '2024-06-01' },
  { number: 'INV-002', client: 'Beta LLC', amount: 1800, status: 'Unpaid', due: '2024-06-15' },
  { number: 'INV-003', client: 'Gamma Inc.', amount: 3200, status: 'Overdue', due: '2024-05-30' },
];
const expenses = [
  { category: 'Salaries', amount: 4000 },
  { category: 'Software', amount: 1200 },
  { category: 'Marketing', amount: 900 },
  { category: 'Travel', amount: 600 },
];
const recentExpenses = [
  { date: '2024-06-10', category: 'Software', amount: 200 },
  { date: '2024-06-09', category: 'Travel', amount: 150 },
];
const cashFlowData = [
  { name: 'Week 1', value: 5000 },
  { name: 'Week 2', value: 3000 },
  { name: 'Week 3', value: 7000 },
  { name: 'Week 4', value: 2000 },
];
const aiInsights = [
  'Upcoming bill: $1,200 due to AWS on 2024-06-20.',
  'Unusual spending detected in Marketing category.',
  'Cash flow risk: Low balance projected in 2 weeks.',
];
const activityFeed = [
  { type: 'payment', note: 'Received $2,500 from Acme Corp', date: '2024-06-10' },
  { type: 'expense', note: 'Paid $200 for software', date: '2024-06-09' },
];

export default function FinanceHome() {
  return (
    <div className="p-8 space-y-8">
      {/* Finance Summary */}
      <div className="grid grid-cols-1 md:grid-cols-4 gap-4">
        <KpiCard title="MTD Revenue" value={`$${financeSummary.mtdRevenue.toLocaleString()}`} />
        <KpiCard title="YTD Revenue" value={`$${financeSummary.ytdRevenue.toLocaleString()}`} />
        <KpiCard title="Cash on Hand" value={`$${financeSummary.cashOnHand.toLocaleString()}`} />
        <KpiCard title="Net Profit" value={`$${financeSummary.netProfit.toLocaleString()}`} />
      </div>
      {/* Cash Flow Chart & Expenses */}
      <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
        <div className="rounded-xl border p-4">
          <h3 className="font-semibold mb-2">Cash Flow</h3>
          <SimpleBarChart data={cashFlowData} />
        </div>
        <div className="rounded-xl border p-4">
          <h3 className="font-semibold mb-2">Top Expenses</h3>
          <ul className="divide-y">
            {expenses.map((e) => (
              <li key={e.category} className="flex justify-between py-2">
                <span>{e.category}</span>
                <span>${e.amount.toLocaleString()}</span>
              </li>
            ))}
          </ul>
        </div>
      </div>
      {/* Invoices Table */}
      <div className="rounded-xl border p-4">
        <h3 className="font-semibold mb-2">Invoices</h3>
        <table className="min-w-full text-sm">
          <thead>
            <tr>
              <th className="text-left p-2">Invoice #</th>
              <th className="text-left p-2">Client</th>
              <th className="text-left p-2">Amount</th>
              <th className="text-left p-2">Status</th>
              <th className="text-left p-2">Due Date</th>
            </tr>
          </thead>
          <tbody>
            {invoices.map((inv) => (
              <tr key={inv.number} className="border-t">
                <td className="p-2">{inv.number}</td>
                <td className="p-2">{inv.client}</td>
                <td className="p-2">${inv.amount.toLocaleString()}</td>
                <td className="p-2">{inv.status}</td>
                <td className="p-2">{inv.due}</td>
              </tr>
            ))}
          </tbody>
        </table>
      </div>
      {/* Recent Expenses & AI Insights */}
      <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
        <div className="rounded-xl border p-4">
          <h3 className="font-semibold mb-2">Recent Expenses</h3>
          <ul className="divide-y">
            {recentExpenses.map((e, i) => (
              <li key={i} className="py-2">
                <span>{e.date}</span> - <span>{e.category}</span> <span className="text-xs text-muted-foreground dark:text-gray-200">${e.amount}</span>
              </li>
            ))}
          </ul>
        </div>
        <div className="rounded-xl border p-4">
          <h3 className="font-semibold mb-2">AI Insights</h3>
          <ul className="list-disc pl-5">
            {aiInsights.map((insight, i) => (
              <li key={i}>{insight}</li>
            ))}
          </ul>
        </div>
      </div>
      {/* Activity Feed */}
      <div className="rounded-xl border p-4">
        <h3 className="font-semibold mb-2">Activity Feed</h3>
        <ul className="divide-y">
          {activityFeed.map((a, i) => (
            <li key={i} className="py-2">
              {a.type === 'payment' ? 'ðŸ’¸' : 'ðŸ§¾'} {a.note} <span className="text-xs text-muted-foreground dark:text-gray-200">({a.date})</span>
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
} 