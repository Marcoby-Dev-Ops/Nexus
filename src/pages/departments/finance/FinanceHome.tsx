import React from 'react';
import { DollarSign, TrendingUp, CreditCard, Receipt, ArrowUpRight, Plus, FileText, Calculator, Banknote } from 'lucide-react';
import { PageTemplates } from '@/components/patterns/PageTemplates';
import { ContentCard } from '@/components/patterns/ContentCard';
import { KpiCard } from '@/components/dashboard/KpiCard';
import { SimpleBarChart } from '@/components/dashboard/SimpleBarChart';

/**
 * @name FinanceHome
 * @description Finance department dashboard for managing invoices, expenses, and financial reports.
 * @returns {JSX.Element} The rendered FinanceHome component.
 */

// Sample data for the finance dashboard
const financeKpiData = [
  { title: 'Total Revenue', value: '$124,580', delta: '+12.5%', trend: 'up' },
  { title: 'Outstanding Invoices', value: '$45,200', delta: '-8.3%', trend: 'down' },
  { title: 'Monthly Expenses', value: '$32,100', delta: '+5.2%', trend: 'up' },
  { title: 'Profit Margin', value: '32.8%', delta: '+2.1%', trend: 'up' },
];

const expenseData = [
  { name: 'Salaries', value: 45000 },
  { name: 'Marketing', value: 12000 },
  { name: 'Operations', value: 8500 },
  { name: 'Software', value: 5200 },
  { name: 'Office', value: 3200 },
];

const revenueData = [
  { name: 'Jan', value: 65000 },
  { name: 'Feb', value: 59000 },
  { name: 'Mar', value: 80000 },
  { name: 'Apr', value: 81000 },
  { name: 'May', value: 56000 },
  { name: 'Jun', value: 124580 },
];

const recentTransactions = [
  { description: 'Client Payment - Acme Corp', amount: '+$45,000', type: 'income', date: '2024-01-15' },
  { description: 'Software Subscription', amount: '-$299', type: 'expense', date: '2024-01-14' },
  { description: 'Marketing Campaign', amount: '-$2,500', type: 'expense', date: '2024-01-13' },
  { description: 'Consulting Fee', amount: '+$8,500', type: 'income', date: '2024-01-12' },
];

const quickActions = [
  { label: 'New Invoice', icon: <Receipt className="w-5 h-5" />, onClick: () => console.log('New Invoice') },
  { label: 'Record Expense', icon: <CreditCard className="w-5 h-5" />, onClick: () => console.log('Record Expense') },
  { label: 'Generate Report', icon: <FileText className="w-5 h-5" />, onClick: () => console.log('Generate Report') },
  { label: 'Calculate Tax', icon: <Calculator className="w-5 h-5" />, onClick: () => console.log('Calculate Tax') },
];

const FinanceHome: React.FC = () => {
  return (
    <PageTemplates.Department
      title="Finance Hub"
      subtitle="Manage invoices, expenses, and financial reports with automated reconciliation"
    >
      {/* Quick Actions */}
      <ContentCard title="Quick Actions" variant="elevated" className="mb-8">
        <div className="grid grid-cols-2 md:grid-cols-4 gap-4">
          {quickActions.map((action, index) => (
            <button
              key={index}
              onClick={action.onClick}
              className="flex flex-col items-center p-6 rounded-lg bg-background border border-border hover:border-primary hover:bg-primary/5 hover:shadow-md transition-all duration-200 group"
            >
              <div className="p-3 rounded-lg bg-primary/10 text-primary group-hover:bg-primary/20 transition-all duration-200 mb-3">
                {action.icon}
              </div>
              <span className="text-sm font-medium text-foreground group-hover:text-primary transition-colors duration-200">
                {action.label}
              </span>
            </button>
          ))}
        </div>
      </ContentCard>

      {/* KPI Cards */}
      <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-6 mb-8">
        {financeKpiData.map((kpi, index) => (
          <KpiCard key={index} {...kpi} />
        ))}
      </div>

      {/* Charts */}
      <div className="grid grid-cols-1 lg:grid-cols-2 gap-8 mb-8">
        <ContentCard 
          title="Revenue Trend" 
          variant="elevated"
          className="hover:shadow-xl transition-all duration-300"
          action={
            <button className="flex items-center space-x-2 px-4 py-4 bg-primary/5 text-primary hover:bg-primary/10 hover:text-primary/90 rounded-lg transition-all duration-200 font-medium border border-border">
              <span className="text-sm">View Details</span>
              <ArrowUpRight className="w-4 h-4" />
            </button>
          }
        >
          <div className="mb-6">
            <p className="text-sm text-muted-foreground">Monthly revenue performance</p>
          </div>
          <SimpleBarChart data={revenueData} />
        </ContentCard>

        <ContentCard 
          title="Expense Breakdown" 
          variant="elevated"
          className="hover:shadow-xl transition-all duration-300"
          action={
            <button className="flex items-center space-x-2 px-4 py-4 bg-primary/5 text-primary hover:bg-primary/10 hover:text-primary/90 rounded-lg transition-all duration-200 font-medium border border-border">
              <span className="text-sm">Manage Expenses</span>
              <ArrowUpRight className="w-4 h-4" />
            </button>
          }
        >
          <div className="mb-6">
            <p className="text-sm text-muted-foreground">Monthly expense categories</p>
          </div>
          <SimpleBarChart data={expenseData} />
        </ContentCard>
      </div>

      {/* Recent Transactions */}
      <ContentCard 
        title="Recent Transactions" 
        variant="elevated"
        action={
          <button className="px-4 py-4 bg-primary/5 text-primary hover:bg-primary/10 hover:text-primary/90 rounded-lg transition-all duration-200 text-sm font-medium border border-border">
            View All Transactions
          </button>
        }
      >
        <div className="space-y-4">
          {recentTransactions.map((transaction, index) => (
            <div key={index} className="flex items-center justify-between p-4 rounded-lg hover:bg-muted/50 transition-colors duration-200">
              <div className="flex items-center space-x-4">
                <div className={`p-4 rounded-lg ${transaction.type === 'income' ? 'bg-success/10 text-success' : 'bg-destructive/10 text-destructive'}`}>
                  {transaction.type === 'income' ? (
                    <DollarSign className="w-5 h-5" />
                  ) : (
                    <CreditCard className="w-5 h-5" />
                  )}
                </div>
                <div>
                  <h4 className="font-medium text-foreground">{transaction.description}</h4>
                  <p className="text-sm text-muted-foreground">{transaction.date}</p>
                </div>
              </div>
              <div className={`font-semibold ${transaction.type === 'income' ? 'text-success' : 'text-destructive'}`}>
                {transaction.amount}
              </div>
            </div>
          ))}
        </div>
      </ContentCard>
    </PageTemplates.Department>
  );
};

export default FinanceHome; 