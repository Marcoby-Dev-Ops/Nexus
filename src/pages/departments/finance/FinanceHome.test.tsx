/**
 * @file FinanceHome.test.tsx
 * @description Unit and snapshot tests for the FinanceHome component.
 */
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import FinanceHome from './FinanceHome';

describe('FinanceHome', () => {
  it('renders finance home title', () => {
    render(<FinanceHome />);
    expect(screen.getByText(/finance/i)).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<FinanceHome />);
    expect(asFragment()).toMatchSnapshot();
  });
}); 