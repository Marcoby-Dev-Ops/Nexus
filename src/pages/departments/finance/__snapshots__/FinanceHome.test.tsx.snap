// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinanceHome matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="min-h-screen bg-gradient-to-br from-gray-50 via-blue-50/30 to-indigo-50/50 dark:from-gray-900 dark:via-gray-900 dark:to-gray-800"
  >
    <div
      class="p-8 space-y-8"
    >
      <div
        class="mb-8"
      >
        <h1
          class="text-3xl font-bold mb-2"
        >
          Finance Hub
        </h1>
        <p
          class="text-lg"
        >
          Manage invoices, expenses, and financial reports with automated reconciliation
        </p>
      </div>
      <div
        class="bg-card dark:bg-background rounded-2xl p-8 shadow-lg border border-border/50 dark:border-border/50 hover:shadow-xl transition-all duration-300 mb-8"
      >
        <div
          class="flex items-center justify-between mb-6"
        >
          <h3
            class="text-xl font-bold text-foreground"
          >
            Quick Actions
          </h3>
        </div>
        <div
          class="grid grid-cols-2 md:grid-cols-4 gap-4"
        >
          <button
            class="flex flex-col items-center p-6 rounded-lg bg-background border border-border hover:border-primary hover:bg-primary/5 hover:shadow-md transition-all duration-200 group"
          >
            <div
              class="p-4 rounded-lg bg-primary/10 text-primary group-hover:bg-primary/20 transition-all duration-200 mb-3"
            >
              <svg
                class="lucide lucide-receipt w-5 h-5"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4 2v20l2-1 2 1 2-1 2 1 2-1 2 1 2-1 2 1V2l-2 1-2-1-2 1-2-1-2 1-2-1-2 1Z"
                />
                <path
                  d="M16 8h-6a2 2 0 1 0 0 4h4a2 2 0 1 1 0 4H8"
                />
                <path
                  d="M12 17.5v-11"
                />
              </svg>
            </div>
            <span
              class="text-sm font-medium text-foreground group-hover:text-primary transition-colors duration-200"
            >
              New Invoice
            </span>
          </button>
          <button
            class="flex flex-col items-center p-6 rounded-lg bg-background border border-border hover:border-primary hover:bg-primary/5 hover:shadow-md transition-all duration-200 group"
          >
            <div
              class="p-4 rounded-lg bg-primary/10 text-primary group-hover:bg-primary/20 transition-all duration-200 mb-3"
            >
              <svg
                class="lucide lucide-credit-card w-5 h-5"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="14"
                  rx="2"
                  width="20"
                  x="2"
                  y="5"
                />
                <line
                  x1="2"
                  x2="22"
                  y1="10"
                  y2="10"
                />
              </svg>
            </div>
            <span
              class="text-sm font-medium text-foreground group-hover:text-primary transition-colors duration-200"
            >
              Record Expense
            </span>
          </button>
          <button
            class="flex flex-col items-center p-6 rounded-lg bg-background border border-border hover:border-primary hover:bg-primary/5 hover:shadow-md transition-all duration-200 group"
          >
            <div
              class="p-4 rounded-lg bg-primary/10 text-primary group-hover:bg-primary/20 transition-all duration-200 mb-3"
            >
              <svg
                class="lucide lucide-file-text w-5 h-5"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V7Z"
                />
                <path
                  d="M14 2v4a2 2 0 0 0 2 2h4"
                />
                <path
                  d="M10 9H8"
                />
                <path
                  d="M16 13H8"
                />
                <path
                  d="M16 17H8"
                />
              </svg>
            </div>
            <span
              class="text-sm font-medium text-foreground group-hover:text-primary transition-colors duration-200"
            >
              Generate Report
            </span>
          </button>
          <button
            class="flex flex-col items-center p-6 rounded-lg bg-background border border-border hover:border-primary hover:bg-primary/5 hover:shadow-md transition-all duration-200 group"
          >
            <div
              class="p-4 rounded-lg bg-primary/10 text-primary group-hover:bg-primary/20 transition-all duration-200 mb-3"
            >
              <svg
                class="lucide lucide-calculator w-5 h-5"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="20"
                  rx="2"
                  width="16"
                  x="4"
                  y="2"
                />
                <line
                  x1="8"
                  x2="16"
                  y1="6"
                  y2="6"
                />
                <line
                  x1="16"
                  x2="16"
                  y1="14"
                  y2="18"
                />
                <path
                  d="M16 10h.01"
                />
                <path
                  d="M12 10h.01"
                />
                <path
                  d="M8 10h.01"
                />
                <path
                  d="M12 14h.01"
                />
                <path
                  d="M8 14h.01"
                />
                <path
                  d="M12 18h.01"
                />
                <path
                  d="M8 18h.01"
                />
              </svg>
            </div>
            <span
              class="text-sm font-medium text-foreground group-hover:text-primary transition-colors duration-200"
            >
              Calculate Tax
            </span>
          </button>
        </div>
      </div>
      <div
        class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-6 mb-8"
      >
        <div
          class="rounded-xl border p-4 shadow-sm flex flex-col gap-1"
        >
          <p
            class="text-muted-foreground mb-1 text-xs uppercase"
          >
            Total Revenue
          </p>
          <div
            class="flex items-end justify-between gap-2"
          >
            <p
              class="text-3xl font-semibold"
            >
              $0.00
            </p>
          </div>
        </div>
        <div
          class="rounded-xl border p-4 shadow-sm flex flex-col gap-1"
        >
          <p
            class="text-muted-foreground mb-1 text-xs uppercase"
          >
            Outstanding Invoices
          </p>
          <div
            class="flex items-end justify-between gap-2"
          >
            <p
              class="text-3xl font-semibold"
            >
              $45,200
            </p>
          </div>
          <p
            class="text-xs font-medium text-destructive"
          >
            -8.3%
          </p>
        </div>
        <div
          class="rounded-xl border p-4 shadow-sm flex flex-col gap-1"
        >
          <p
            class="text-muted-foreground mb-1 text-xs uppercase"
          >
            Monthly Expenses
          </p>
          <div
            class="flex items-end justify-between gap-2"
          >
            <p
              class="text-3xl font-semibold"
            >
              $32,100
            </p>
          </div>
          <p
            class="text-xs font-medium text-success"
          >
            +5.2%
          </p>
        </div>
        <div
          class="rounded-xl border p-4 shadow-sm flex flex-col gap-1"
        >
          <p
            class="text-muted-foreground mb-1 text-xs uppercase"
          >
            Profit Margin
          </p>
          <div
            class="flex items-end justify-between gap-2"
          >
            <p
              class="text-3xl font-semibold"
            >
              32.8%
            </p>
          </div>
          <p
            class="text-xs font-medium text-success"
          >
            +2.1%
          </p>
        </div>
      </div>
      <div
        class="grid grid-cols-1 lg:grid-cols-2 gap-8 mb-8"
      >
        <div
          class="bg-card dark:bg-background rounded-2xl p-8 shadow-lg border border-border/50 dark:border-border/50 hover:shadow-xl transition-all duration-300 hover:shadow-xl transition-all duration-300"
        >
          <div
            class="flex items-center justify-between mb-6"
          >
            <h3
              class="text-xl font-bold text-foreground"
            >
              Revenue Trend
            </h3>
            <div
              class="flex-shrink-0"
            >
              <button
                class="flex items-center space-x-2 px-4 py-4 bg-primary/5 text-primary hover:bg-primary/10 hover:text-primary/90 rounded-lg transition-all duration-200 font-medium border border-border"
              >
                <span
                  class="text-sm"
                >
                  View Details
                </span>
                <svg
                  class="lucide lucide-arrow-up-right w-4 h-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7 7h10v10"
                  />
                  <path
                    d="M7 17 17 7"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="mb-6"
          >
            <p
              class="text-sm text-muted-foreground"
            >
              Monthly revenue performance
            </p>
          </div>
          <div
            class="recharts-responsive-container"
            style="width: 100%; height: 200px; min-width: 0;"
          />
        </div>
        <div
          class="bg-card dark:bg-background rounded-2xl p-8 shadow-lg border border-border/50 dark:border-border/50 hover:shadow-xl transition-all duration-300 hover:shadow-xl transition-all duration-300"
        >
          <div
            class="flex items-center justify-between mb-6"
          >
            <h3
              class="text-xl font-bold text-foreground"
            >
              Expense Breakdown
            </h3>
            <div
              class="flex-shrink-0"
            >
              <button
                class="flex items-center space-x-2 px-4 py-4 bg-primary/5 text-primary hover:bg-primary/10 hover:text-primary/90 rounded-lg transition-all duration-200 font-medium border border-border"
              >
                <span
                  class="text-sm"
                >
                  Manage Expenses
                </span>
                <svg
                  class="lucide lucide-arrow-up-right w-4 h-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7 7h10v10"
                  />
                  <path
                    d="M7 17 17 7"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="mb-6"
          >
            <p
              class="text-sm text-muted-foreground"
            >
              Monthly expense categories
            </p>
          </div>
          <div
            class="recharts-responsive-container"
            style="width: 100%; height: 200px; min-width: 0;"
          />
        </div>
      </div>
      <div
        class="bg-card dark:bg-background rounded-2xl p-8 shadow-lg border border-border/50 dark:border-border/50 hover:shadow-xl transition-all duration-300 "
      >
        <div
          class="flex items-center justify-between mb-6"
        >
          <h3
            class="text-xl font-bold text-foreground"
          >
            Recent Transactions
          </h3>
          <div
            class="flex-shrink-0"
          >
            <button
              class="px-4 py-4 bg-primary/5 text-primary hover:bg-primary/10 hover:text-primary/90 rounded-lg transition-all duration-200 text-sm font-medium border border-border"
            >
              View All Transactions
            </button>
          </div>
        </div>
        <div
          class="space-y-4"
        >
          <p
            class="text-sm text-muted-foreground"
          >
            No PayPal transactions found.
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
